### PR TITLE
Refactor(styling): Add redux-saga and update fetch collection with sa…

### PR DESCRIPTION
**Pull request by Arjunmdas3211**
…ga-middleware

### CODE REVIEW

File: debug.log [hunk 1]
```diff
@@ -4,3 +4,4 @@
 [1014/151835.329:ERROR:directory_reader_win.cc(43)] FindFirstFile: The system cannot find the path specified. (0x3)
```
Comment (amouxkin) [inline debug.log:4]: Avoid committing `log`, `lock`, files.

File: src/pages/shop/shop.component.jsx [hunk 2]
```diff
@@ -21,8 +21,8 @@ class ShopPage extends React.Component {
   };
 
   componentDidMount() {
-    const { fetchCollectionsStartAsync } = this.props
-    fetchCollectionsStartAsync();
+    const { fetchCollectionsStart } = this.props;
```
Comment (amouxkin) [inline src/pages/shop/shop.component.jsx:24]: Avoid unnecessary restructuring.

File: src/redux/shop/shop.sagas.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { takeEvery, call, put } from 'redux-saga/effects';
+
+import { firestore, convertCollectionsSnapshotToMap } from '../../firebase/firebase.utils';
+import {fetchCollectionsSuccess, fetchCollectionsFailure } from './shop.actions';
+
+import ShopActionTypes from './shop.types';
+
+export function* fetchCollectionsAsync() {
+  try{
```
Comment (amouxkin) [inline src/redux/shop/shop.sagas.js]: Try to run prettier/eslint before PR.

File: debug.log (hunk 1 above)
Comment (Arjunmdas3211) [reply debug.log:4]: ok i'll do

**Pull request merged**